src/components/Modals/Receive/Receive.tsx

```diff
@@ -11,97 +11,153 @@ import {
         <ModalHeader textAlign='center'>
-          {translate('modals.receive.receiveAsset', { asset: 'Bitcoin' })}
+          {translate('modals.receive.receiveAsset', { asset: name })}
         </ModalHeader>
         <ModalCloseButton />
-        <ModalBody alignItems='center' justifyContent='center'>
-          <Card variant='inverted' width='auto' borderRadius='xl'>
-            <Card.Body>
-              <QRCode text='3F9geWenC9PiUzmmSwLanjSqXLBAQfaryz' />
-            </Card.Body>
-            <Card.Footer textAlign='center' pt={0}>
-              <RawText>3F9geWenC9PiUzmmSwLanjSqXLBAQfaryz</RawText>
-            </Card.Footer>
-          </Card>
-        </ModalBody>
-        <ModalFooter flexDir='column'>
-          <Box>
-            <Text
-              translation={['modals.receive.onlySend', { asset: 'Bitcoin', symbol: 'BTC' }]}
-              color='gray.500'
-              textAlign='center'
-            />
-          </Box>
-          <HStack my={6} spacing={8}>
-            <Button
-              color='gray.500'
-              flexDir='column'
-              role='group'
-              variant='link'
-              _hover={{ textDecoration: 'none', color: useColorModeValue('gray.900', 'white') }}
-            >
-              <Circle
-                bg={useColorModeValue('gray.100', 'gray.700')}
-                mb={2}
-                size='40px'
-                _groupHover={{ bg: 'blue.500', color: 'white' }}
-              >
-                <CopyIcon />
-              </Circle>
-              <Text translation='modals.receive.copy' />
-            </Button>
-            <Button
-              color='gray.500'
-              flexDir='column'
-              role='group'
-              variant='link'
-              _hover={{ textDecoration: 'none', color: useColorModeValue('gray.900', 'white') }}
-            >
-              <Circle
-                bg={useColorModeValue('gray.100', 'gray.700')}
-                mb={2}
-                size='40px'
-                _groupHover={{ bg: 'blue.500', color: 'white' }}
-              >
-                <CopyIcon />
-              </Circle>
-              <Text translation='modals.receive.setAmount' />
-            </Button>
-            <Button
-              color='gray.500'
-              flexDir='column'
-              role='group'
-              variant='link'
-              _hover={{ textDecoration: 'none', color: useColorModeValue('gray.900', 'white') }}
-            >
-              <Circle
-                bg={useColorModeValue('gray.100', 'gray.700')}
-                mb={2}
-                size='40px'
-                _groupHover={{ bg: 'blue.500', color: 'white' }}
-              >
-                <ViewIcon />
-              </Circle>
-              <Text translation='modals.receive.verify' />
-            </Button>
-          </HStack>
-        </ModalFooter>
+        {state.wallet ? <WalletConnected /> : <ConnectAWallet />}
       </ModalContent>
     </Modal>
   )
```

Comment:
My only comment here is we are still trying to figure out what is exceptable for our standards. In our documentation we do mention if you are approaching 150 lines or going over maybe we need to think about refactoring. The other is that we are trying to not define multiple components in the same file so we can keep these smaller and more digestible. Also is defining those components in the render without any useMemos causing any issues every time it rerenders?

So Verify, WalletConnect and WalletConnected should those be pulled out?

Reply:
yep - happy to break these out to separate components or memoize them. i've used this pattern before and i just find it cognitively easier than a big ternaries and a jsx triangle of doom.

would you be ok with memoizing within the component? if these components are never used again i think it's ok to scope them to the component.

Reply:
Yea I think that memorizing would be fine so it keeps it a little lighter.